### PR TITLE
feat: add JSONL log reader and detection logging improvements

### PR DESCRIPTION
**Pull request by tphakala**
## Summary

- Add new `internal/logger/reader` package for parsing JSONL log files with date, level, operation, and module filtering, plus deduplication and sorting across multiple files
- Add `GetOutputPath()` method to `CentralLogger` for resolving configured log file paths by module
- Promote `flush_detection` log from DEBUG to INFO so detection timeouts appear in `actions.log`

Prerequisite for the Events page API handlers (`docs/plans/2026-02-28-events-api-handlers.md`).

## Test plan
- [x] 14 unit tests for reader package covering filtering, deduplication, sorting, malformed lines, edge cases
- [x] All tests pass with `-race` flag
- [x] `golangci-lint` passes with 0 issues
- [x] Code reviewed by Gemini (2 rounds) — all findings addressed

## Summary by CodeRabbit

* **New Features**
  * Log reader: read JSONL logs with filtering (date, level, module prefix, operations), preserve extra fields, discover rotated files, merge multiple files with deduplication and chronological ordering
  * Query module-specific log output paths

* **Improvements**
  * Increased logging verbosity for flushed-detection messages (now logged at Info)

* **Tests**
  * Comprehensive tests covering filtering, parsing, deduplication, sorting, file discovery, and malformed lines

### CODE REVIEW

File: internal/logger/reader/reader.go [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+	deduplicated := make([]LogEntry, 0, len(allEntries))
+
+	for i := range allEntries {
+		key := deduplicationKey(&allEntries[i])
+		if _, exists := seen[key]; exists {
+			continue
+		}
+		seen[key] = struct{}{}
+		deduplicated = append(deduplicated, allEntries[i])
+	}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/logger/reader/reader.go:126]: Implemented in-place deduplication. Also eliminates the separate `deduplicated` slice allocation.

File: internal/logger/reader/reader.go [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+		if !knownFields[k] {
+			extra[k] = v
+		}
+	}
+	if len(extra) > 0 {
+		entry.Fields = extra
+	}
+
+	return entry, true
+}
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/logger/reader/reader.go:227]: Refactored to single-pass delete-from-map approach. Removes the `knownFields` global and also preserves non-string values for known field names in `Fields` map instead of silently dropping them.
